.github/workflows/setup-tofu.yml

```diff
@@ -0,0 +1,314 @@
+name: 'Setup OpenTofu'
+
+on:
+  push:
+    branches:
+    - main
+  pull_request:
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  tofu-versions:
+    name: 'OpenTofu Versions'
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+        # TODO: add move versions to test when the feature is added
+        tofu-versions: [latest]
+    steps:
+    - name: Checkout
+      uses: actions/checkout@3df4ab11eba7bda6032a0b82a6bb43b11571feac # v4.0.0
+
+    - name: Setup OpenTofu - ${{ matrix['tofu-versions'] }}
+      uses: ./
+      with:
+        tofu_version: ${{ matrix['tofu-versions'] }}
+
+    - name: Validate OpenTofu Version - ${{ matrix['tofu-versions'] }}
+      if: ${{ matrix['tofu-versions'] != 'latest' }}
+      run: tofu version | grep ${{ matrix['tofu-versions']}}
+
+    - name: Validate OpenTofu Version - ${{ matrix['tofu-versions'] }}
+      if: ${{ matrix['tofu-versions'] == 'latest' }}
+      run: tofu version | grep 'OpenTofu v'
+
+  tofu-versions-no-wrapper:
```

Comment:
who is the wrapper?
seems to be the same definition as above?

Reply:
It does seem so, however, the differ: the `tofu-versions` uses the action with the `wrapper` activated, the `tofu-versions-no-wrapper` uses the actions with the `wrapper` turned off. The `wrapper`, as the name suggests, is a process which wraps the tofu process to capture its stdout/stderr and the exit code.